weasyl/controllers/user.py

```diff
@@ -139,12 +143,14 @@ def signin_2fa_auth_get_(request):
 
 @guest_required
 @token_checked
+@rate_limit_route(expiration_time=60 * 1)
 def signin_2fa_auth_post_(request):
     sess = define.get_weasyl_session()
+    referer = request.params["referer"] if 'referer' in request.params else "/"
 
     # Only render page if the session exists //and// the password has
     # been authenticated (we have a UserID stored in the session)
-    if not sess.additional_data or '2fa_pwd_auth_userid' not in sess.additional_data:
+    if not sess.additional_data and '2fa_pwd_auth_userid' not in sess.additional_data:
```

Comment:
This change doesn’t look right. Note that `not` has a higher precedence than `and`.